From e08be7fb6d260e51462d0d116e131a3527120a3d Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Sun, 29 Jul 2018 20:19:19 +0200
Subject: [PATCH 11/13] LineageParts: Hide dark mode on light design

* If light design is chosen it doesn't make sense to choose the preferred
  dark mode
* Hide the preference in that case

Change-Id: I0ef64afe91abd52572906c3eb280fec2a4592cad
---
 .../lineageparts/style/StylePreferences.java         | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/org/lineageos/lineageparts/style/StylePreferences.java b/src/org/lineageos/lineageparts/style/StylePreferences.java
index 2907fae..e45c875 100644
--- a/src/org/lineageos/lineageparts/style/StylePreferences.java
+++ b/src/org/lineageos/lineageparts/style/StylePreferences.java
@@ -61,6 +61,7 @@ public class StylePreferences extends SettingsPreferenceFragment {
 
     private Preference mStylePref;
     private Preference mAccentPref;
+    LineageSystemSettingListPreference mDarkPref;
 
     private List<Accent> mAccents;
 
@@ -88,9 +89,10 @@ public class StylePreferences extends SettingsPreferenceFragment {
         mAccentPref.setOnPreferenceClickListener(this::onAccentClick);
         setupAccentPref();
 
-        LineageSystemSettingListPreference darkPref = (LineageSystemSettingListPreference)
+        mDarkPref = (LineageSystemSettingListPreference)
                 findPreference("berry_dark_overlay");
-        darkPref.setOnPreferenceChangeListener(this::onDarkChange);
+        mDarkPref.setOnPreferenceChangeListener(this::onDarkChange);
+        setDarkStyleVisibility(mInterface.getGlobalStyle());
 
         Preference automagic = findPreference("style_automagic");
         automagic.setOnPreferenceClickListener(p -> onAutomagicClick());
@@ -276,10 +278,16 @@ public class StylePreferences extends SettingsPreferenceFragment {
         // selection dialog to be dismissed gracefully
         new Handler().postDelayed(() -> mInterface.setGlobalStyle(value, mPackageName), 500);
 
+        setDarkStyleVisibility(value);
         setStyleIcon(value);
+
         return true;
     }
 
+    private void setDarkStyleVisibility(int value) {
+        mDarkPref.setVisible(value != StyleInterface.STYLE_GLOBAL_LIGHT);
+    }
+
     private void setStyleIcon(int value) {
         int icon;
         switch (value) {
-- 
2.17.1

